src/engine/Flow.ts

```diff
@@ -0,0 +1,108 @@
+import { CallStack } from "./CallStack";
+import { Choice } from "./Choice";
+import { JsonSerialisation } from "./JsonSerialisation";
+import { InkObject } from "./Object";
+import { SimpleJson } from "./SimpleJson";
+import { Story } from "./Story";
+import { throwNullException } from "./NullException";
+
+export class Flow {
+  public name: string;
+  public callStack: CallStack;
+  public outputStream: InkObject[];
+  public currentChoices: Choice[];
+
+  constructor(name: String, story: Story);
+  constructor(name: String, story: Story, jObject: Record<string, any>);
+  constructor() {
+    let name = arguments[0] as string;
+    let story = arguments[1] as Story;
+
+    this.name = name;
+    this.callStack = new CallStack(story);
+
+    if (arguments[3]) {
+      let jObject = arguments[3] as Record<string, any>;
+
+      this.outputStream = JsonSerialisation.JArrayToRuntimeObjList(
```

Comment:
I'm not 100% sure, but it seems like in this signature, we also need to do `this.callStack.SetJsonToken ((Dictionary < string, object > )jObject ["callstack"], story);` (l. 21 of Flow.cs https://github.com/inkle/ink/compare/0.9.0%E2%80%A6v1.0.0#diff-8cbd10a71465fea0e316355215ad5788ea62c3094e5b1e27d8e99eff1c1ee9f1R21). I think we may be missing that here?

Reply:
Oh yeah, you're right, good catch! I'll fix that.